core/metacling/src/TCling.cxx

```diff
@@ -1456,8 +1456,36 @@ static const char *FindLibraryName(void (*func)())
       // Not in a known share library, let's give up
       return 0;
    } else {
-      //fprintf(stdout,"Found address in %s\n",info.dli_fname);
+      if (strchr(info.dli_fname, '/'))
+         return info.dli_fname;
+      // Else absolute path. For all we know that's a binary.
+      // Some people have dictionaries in binaries, this is how we find their path:
+      // (see also https://stackoverflow.com/a/1024937/6182509)
+# if defined(R__MACOSX)
+      static char buf[PATH_MAX] = { 0 };
```

Comment:
Is this OK not being `thread_local`?

Reply:
In practice it should be okay.  FindLibraryName is static function called only from RegisterModule which should be only called during the initialization part of a library load and this should (ought to?) be protected by the dlopen lock [to be verified though]

Reply:
No need for keeping that static anyway. Thanks!